magnet/helm_inference.py

```diff
@@ -0,0 +1,122 @@
+import contextlib
+import os
+
+from helm.common.request import Request
+from helm.common.request import RequestResult, GeneratedOutput
+from helm.common.authentication import Authentication
+from helm.benchmark.executor import ExecutionSpec, Executor, ExecutorError
+from helm.benchmark.config_registry import (
+    # register_configs_from_directory,
+    register_builtin_configs_from_helm_package,
+)
+from helm.common.hierarchical_logger import hwarn
+from helm.clients.huggingface_client import HuggingFaceServerFactory
+
+
+class HelmInferenceEngine:
+    """
```

Comment:
If you have a "\n" character anywhere in the doctest, then you will want to make this a "raw" r-string to prevent python from interpreting it as a literal newline on the docstring level. In general it is always safe to make a docstring an "r-string".